src/app/modules/startup/internal/state_factory.nim

```diff
@@ -0,0 +1,134 @@
+import chronicles
+import ../../../../app_service/service/keycard/constants
+import ../controller
+from ../../../../app_service/service/keycard/service import PINLengthForStatusApp
+from ../../../../app_service/service/keycard/service import PUKLengthForStatusApp
+import state
+
+logScope:
+  topics = "startup-module-state-factory"
+
+# Forward declaration
+proc createState*(stateToBeCreated: StateType, flowType: FlowType, backState: State): State
+
+include biometrics_state 
+include keycard_create_pin_state 
+include keycard_display_seed_phrase_state 
+include keycard_empty_state
+include keycard_enter_pin_state 
+include keycard_enter_puk_state 
+include keycard_enter_seed_phrase_words_state 
+include keycard_insert_keycard_state
+include keycard_locked_state 
+include keycard_max_pairing_slots_reached_state 
+include keycard_max_pin_retries_reached_state
+include keycard_max_puk_retries_reached_state 
+include keycard_not_empty_state 
+include keycard_pin_set_state 
+include keycard_plugin_reader_state 
+include keycard_reading_keycard_state
+include keycard_recover_state
+include keycard_repeat_pin_state 
+include keycard_wrong_pin_state 
+include keycard_wrong_puk_state 
+include notification_state 
+include user_profile_chat_key_state 
+include user_profile_confirm_password_state 
+include user_profile_create_password_state
+include user_profile_create_state 
+include user_profile_enter_seed_phrase_state 
+include user_profile_import_seed_phrase_state 
+include welcome_state_new_user
+include welcome_state_old_user
+include welcome_state
+include login_state
+include login_keycard_insert_keycard_state
+include login_keycard_reading_keycard_state
+include login_keycard_enter_pin_state 
+include login_keycard_wrong_keycard
+include login_keycard_wrong_pin_state
+include login_keycard_max_pin_retries_reached_state
+include login_keycard_max_puk_retries_reached_state
+include login_keycard_empty_state
+
+proc createState*(stateToBeCreated: StateType, flowType: FlowType, backState: State): State =
+  if stateToBeCreated == StateType.AllowNotifications:
```

Comment:
Basically it's the same, since each block returns from this proc I opted for `if`, but even the `switch` is used, in the same order, we will end up with the same number of comparing.